Various/buyone_Convert custom actions to Lua ReaScript/BuyOne_Convert custom actions in batch to Lua ReaScripts or dump as Lua code.lua

```diff
@@ -0,0 +1,390 @@
+	-- 3. Re-initialize the original table in a sorted order, joining the split stings back together
+	local lines_t = {}
+	for k,v in next, tmp_t do
+	lines_t[k] = table.concat(v)
+	end
+
+
+-- Re-order the table by sections: Main, Media Explorer, MIDI Editor, MIDI Events list
+
+-- Construct a temp. table in the pre-define order of the Action list section codes
+local captures_t = {'0','100','32063','32060','32061'}
+local tmp_t = {}
+	for _, sect in ipairs(captures_t) do
+		-- Bring custom actions from the same section together, moving their table values in a predefined order of section codes to a temporary table
+		for _,str in next, lines_t do
+		if str:match('%s('..sect..')%s\"') then tmp_t[#tmp_t+1] = str end
+		end
+	end
+
+
+local lines_t = tmp_t
+local tmp_t = nil
+
+--if converting to individual Lua files
+local resp = flags_t[3] and r.MB('There\'re '..tostring(#lines_t)..' custom actions to export.','INFO',1)
+		if resp == 2 then return end
+
+
+-- Concatenate the path
+	if subdir ~= '' and subdir ~= 'existing or new; or leave as it is or empty' then
+	-- check if such directory already exists and if not, create
+		local i = 0
+			repeat dir = r.EnumerateSubdirectories(path..sep..'Scripts'..sep, i)
+				if dir == subdir then break end
+			i = i + 1
+			until not dir
+		if not dir then r.RecursiveCreateDirectory(path..sep..'Scripts'..sep..subdir, 0) -- should return 0 on failure but i prefer explicit confirmation below
+		-- check if the directory has been created
+		local t_point = r.time_precise() -- wait for .5 a sec until the function cache is cleared, otherwise the created dir isn't registered
+		repeat until r.time_precise() - t_point > 0.5
+		local i = 0
+			repeat dir = r.EnumerateSubdirectories(path..sep..'Scripts'..sep, i)
+				if dir == subdir then break end
+			i = i + 1
+			until not dir
+		end
+		if dir then f_path = path..sep..'Scripts'..sep..subdir
+		subdir_txt = '\\'..subdir -- for the concluding message in the end
+		else f_path = path..sep..'Scripts'; subdir_txt = ''
+		r.MB('      Folder creation has failed.\n\nThe exported files will be placed\n\n       in the\\Scripts  directory.','WARNING',0)
+		end
+	else f_path = path..sep..'Scripts'; subdir_txt = '' end
+
+
+-- Concatenate a txt dump file extension and name
+	if flags_t[1] and flags_t[2] then inset = 'Main & MIDI '
+	elseif flags_t[1] then inset = 'Main '
+	elseif flags_t[2] then inset = 'MIDI ' end
+	if flags_t[1] and flags_t[2] and flags_t[5] then inset = 'Main & MIDI from Arrange '
+	elseif flags_t[5] then inset = 'MIDI from Arrange ' end
+
+local pref, f_name, ext = '', inset..tostring(#lines_t)..' custom actions LUA dump', '.txt' -- name pattern for individual LUA files is concatenated inside the MAIN LOOP since it depends on each custom action properties
+
+
+-- Check if there's already a txt dump file in the target directory and ovewrite if confirmed by the user
+	if flags_t[4] then
+		for f in io.popen('dir \"'..f_path..sep..'\" /b'):lines() do
```

Comment:
Another unprotected `io.popen` here. A simple way to check whether a file exists and is readable:

```lua
function file_exists(fn)
   local f = io.open(fn, 'r')
   return f ~= nil and f:close()
end

file_exists(f_path..sep..f)
```

Reply:
Oops, will fix.

Reply:
Done.
Could you clarify the line `return f ~= nil and f:close()`? Does it return true if the `f` isn't nil? And is it a kind of ternary operator where the file is closed only if `f` is NOT nil? And if it's nil, is it not closed or is there simply nothing to close?

Reply:
All of the above. It returns true only if `f` is not nil and it was closed successfully. The `f:close()` is skipped if `f` is nil.

It's the same as:
```lua
if f ~= nil then
  return f:close()
else
  return false
end
```

Reply:
Got it, thank you.